### PR TITLE
adds fieldoffset, fixes #218

### LINKED ISSUE #218: Provide `fieldoffset`
Julia 0.4 has `fieldoffsets`, returning an array.
Julia 0.5 has `fieldoffset`, returning a single type.

### PR DESCRIPTION
**Pull request by ssfrr**

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -347,6 +347,11 @@ if VERSION < v"0.4.0-dev+3609"
     export fieldnames
 end
 
+if VERSION < v"0.5.0-dev+2285"
+    fieldoffset(T, i) = fieldoffsets(T)[i]
```
Comment (TotalVerb) [inline src/Compat.jl]: This returns the wrong type. Should probably be converted to `UInt` before returning.

File: test/runtests.jl [hunk 2]
```diff
@@ -267,6 +267,8 @@ if VERSION < v"0.4.0-dev+3609"
     end
 end
 
+@test fieldoffset(Complex{Float32}, 2) == 4
```
Comment (TotalVerb) [inline test/runtests.jl]: Test `===` with `Unsigned(4)`?

**Comment by ssfrr** [comment]
Thanks for the review, @TotalVerb. I updated the test and fixed the implementation to return a `UInt`.

**Comment by TotalVerb** [comment]
LGTM

**Comment by tkelman** [comment]
Needs README entry.

**Comment by stevengj** [comment]
@tkelman, the PR has a README entry.

**Comment by tkelman** [comment]
whoops, sorry, not sure why I thought it didn't

**Pull request merged**